packages/flutter_tools/lib/src/cache.dart

```diff
@@ -280,10 +340,15 @@ abstract class CachedArtifact {
   }
 
   /// Hook method for extra checks for being up-to-date.
-  bool isUpToDateInner() => true;
+  bool isUpToDateInner({BuildMode buildMode, TargetPlatform targetPlatform, bool skipUnknown}) => true;
```

Comment:
Rather than having `isUpToDateInner()`, why not just have subclasses override `isUpToDate()` and say:

```dart
  UpdateResult result = super.isUpToDate(...);
  if (!result.isUpToDate)
    return result;
```

Reply:
Addressed in linked PR